src/components/Checkbox/hooks/useFirefoxShiftLabelSupport.jsx

```diff
@@ -0,0 +1,23 @@
+import { useCallback, useMemo } from "react";
+import { detect } from "detect-browser";
+
+const BROWSER_NAME = "firefox";
+
+export function useFirefoxShiftLabelSupport({ inputRef }) {
+  const isFirefox = useMemo(() => {
+    const browser = detect();
```

Comment:
this can be done in the scope level - i don't think that the browser will change ?